### DISCUSSION TITLE
API returns image on localhost but not when deployed on Vercel

### DISCUSSION THREAD

**Discussion by DovileUnrvld** [vercel/next.js#52105] [Help] [answered]
### Summary

Hi all, I am trying to set up an API for fetching images from third party source and serving them as a NextJS API.

My serverless function:
```javascript
const handler = async (req: NextApiRequest, res: NextApiResponse) => {
  const { id } = req.query;

  try {
    const { url } = getURL(id);

    if (!url) return res.status(404).end();

    const imageResponse = await fetch(url);

    if (!imageResponse.ok) return res.status(404).end();

    const contentType = imageResponse.headers.get("content-type");
    const body = imageResponse .body;

    res.setHeader('content-type', contentType || 'image/webp');

    return res.status(200).send(body);
  } catch (error) {
    console.log('GET IMAGE API ERROR', error);
    return res.status(404).end();
  }
};

export default handler;

export const config = {
  api: {
    externalResolver: true
  }
};
``` 

All I get in response for deployed version is empty object {}. On localhost it works fine.

I am using Nextjs 13.1 version.

**Comment by icyJoseph** [comment, marked as answer]
What do you get if you deploy with a log `console.log(typeof body)` before the return?

Also, what Node version have you got locally, and on Vercel? they should match.

If that still fails, please take a look here: https://github.com/vercel/next.js/discussions/41050#discussioncomment-3770024

**Reply by DovileUnrvld** [reply]
Thank you so much! I had different versions when I thought without checking they are the same...
I managed to replicate issue locally and then fixed it by doing:
```javascript
   const readableStream = response.body as unknown as NodeJS.ReadableStream;
   return res.status(200).send(Readable.from(readableStream));
```
